### PR TITLE
Attempt to fix #1199

### LINKED ISSUE #1199: Go-to-definition of a module defined using a functor leads to functor not module
Go-to-definition of a module defined using a functor shows the functor used to define the module instead of the place where the module was defined. 

Example:

```ocaml
1 module Make () = struct
2   let u = ()
3 end
4
5 module T = Make ();;
6 
7 let () = T.u
```

Go-to-definition on `T` in `T.u` in goes to the definition of `Make` (line 1) instead of `T` (on line 5). 

I think that's not the expected behavior.

Using the (latest) ocamllsp 1.2.0-5-g68a0237. (I think that ocaml-lsp relies on merlin for this feature, so reporting here)

### PR DESCRIPTION
**Pull request by rgrinberg**
Not confident this is correct, but it should be easy to evaluate since it's a tiny patch.

### CODE REVIEW

File: src/analysis/typedtrie.ml [hunk 1]
```diff
@@ -677,15 +677,19 @@ let rec follow ~remember_loc ~state scopes ?before trie path =
...
-              { state with
-                functor_arguments = functor_argument :: state.functor_arguments
-              }
-            in
-            inspect_functor (Namespaced_path.peal_head_exn path) state
-              (snd funct)
+            let path = Namespaced_path.peal_head_exn path in
+            if Namespaced_path.equal path Namespaced_path.empty then
+              Found (loc, doc)
+            else begin
```
Comment (trefis) [inline src/analysis/typedtrie.ml:683]: Just for consistency with the cases above, I'd write this as:
```ocaml
let path = Namespaced_path.peal_head_exn path in
begin match Namespaced_path.head path with
| None -> Found (loc, doc)
| Some _ ->
```

**Review by trefis — APPROVED** [review]
That looks correct indeed!

File: src/analysis/typedtrie.ml (hunk 1 above)
Comment (rgrinberg) [reply src/analysis/typedtrie.ml]: done!

**Comment by voodoos** [comment]
Thank you ! Can you add the changelog entry ?

**Comment by rgrinberg** [comment]
Added!

**Pull request merged**